src/pcre2_compile.c

```diff
@@ -1907,7 +1908,7 @@ else
       are octal escapes if there are not that many previous captures. */
 
       if (read_number(&ptr, ptrend, -1, INT_MAX/10 - 1, 0, &s, errorcodeptr) &&
-          (s < 10 || oldptr[-1] >= CHAR_8 || s <= (int)cb->bracount))
+          (s < 10 || oldptr[-1] >= CHAR_8 || (unsigned)s <= bracount))
```

Comment:
minor nitpick: use uint32_t instead of unsigned

Reply:
I think `unsigned` is better. The variable `s` is declared as `int`, and it is known to be non-negative, so the correct conversion to widen it to the corresponding unsigned type is `unsigned`. On a (hypothetical) architecture with 64-bit `int`, casting to `uint32_t` would be incorrect.